### PR TITLE
feature/68 - fix dateHistogram and test

### LINKED ISSUE #68: fix dateHistogram test
looks like the component itself is doing $addToSet when it ought to be $sum, and the test is unfinished/inaccurate given those difficulties.

### PR DESCRIPTION
**Pull request by doug-patterson**
Fixes #68 

### Description
* No idea what happened, but neither dateHistogram nor its test were right on master.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>This PR has failing tests. Please alleviate the errors and commit them

<details>
  <summary>result should output the expected query and results</summary>
  <code>expected { Object (entries) } to deeply equal { Object (entries) }</code>
</details></td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Your PR has lint warnings. Please consider fixing these.
      
<details>
  <summary>dateHistogram.js</summary>
  <p><a href="https://github.com/smartprocure/contexture-mongo/blob/feature/68/src/example-types/dateHistogram.js" target="_blank">src/example-types/dateHistogram.js</a></p>
  
  <strong>'x' is defined but never used.</strong>
  <code>Line 49: ...keysToObject(x => 1, include),</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 430397d295ca65ca6fd48f23b1659a84c9ffa55c
</p>

**Comment by doug-patterson** [comment]
@Akrion `npm run test` and `npm run test:ci` both pass everything locally. I'm stumped. Any ideas?

**Comment by doug-patterson** [comment]
seems to be some issue running the tests at all if you go find it on circle:

```
> contexture-mongo@0.11.3 test:ci /home/circleci/project
> ./node_modules/mocha/bin/_mocha --recursive --reporter json > test-results.json

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! contexture-mongo@0.11.3 test:ci: `./node_modules/mocha/bin/_mocha --recursive --reporter json > test-results.json`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the contexture-mongo@0.11.3 test:ci script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/circleci/.npm/_logs/2020-03-18T20_28_40_494Z-debug.log

Exited with code exit status 1
```

**Comment by doug-patterson** [comment]
tests also pass as low as node 8.9.0, Circle is on latest 9

**Comment by doug-patterson** [comment]
ok, it really is supposedly my code since taking out the test gets me a pass. But test:ci passes that code locally.

**Comment by doug-patterson** [comment]
OK, duti is now in an error state, but it's not reporting an error on any test.

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>This PR has failing tests. Please alleviate the errors and commit them

<details>
  <summary>result should output the expected query and results</summary>
  <code>expected { Object (entries) } to deeply equal { Object (entries) }</code>
</details></td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2b9447e03cf92ca521fd028ea4a14ab8a4cc3f7a
</p>

**Comment by doug-patterson** [comment]
OK, I actually think I get it: it is my test, and it's a timezone issue that's making the expected keys on the output data different when you run it in different timezones

**Comment by doug-patterson** [comment]
@stellarhoof @Akrion figured it out, Circle was fine, the keys on the histogram are timestamps so they were different when it ran on CI than when you run it locally. Altered the test to not be sensitive to timezone. Ready for review.

**Comment by Akrion** [comment]
@doug-patterson awesome. Great detective work 👍

**Review by Akrion — APPROVED** [review]

**Comment by doug-patterson** [comment]
@Akrion I don't have merge permissions on this repo -- could you merge unless you want a second opinion?

**Comment by Akrion** [comment]
@doug-patterson yeah I was hoping someone else would also take a look as well just in case. It has been awhile since I dealt with this repo.

**Review by dshishkov — APPROVED** [review]
Discussed with Doug some use cases for cardinality in this type, currently we are not using anywhere but the suggested change seems to have valid use cases and fixes issues with the existing tests.

**Comment by doug-patterson** [comment]
sounds like we're good to go -- thanks for looking.

**Comment by doug-patterson** [comment]
also, the companion PRs are tiny and could use a look if you've got the time

File: test/example-types/dateHistogram.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ describe('dateHistogram', () => {
             min: { $min: '$metrics.usersCount' },
             sum: { $sum: '$metrics.usersCount' },
             count: { $sum: 1 },
-            cardinality: { $addToSet: '$metrics.usersCount' },
+            cardinality: { $sum: '$metrics.usersCount' },
```
Comment (daedalus28) [inline test/example-types/dateHistogram.js]: cardinality and sum are not the same thing - this introduces a bug

**Pull request merged**